cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -433,3 +433,18 @@ endif()
 # complain about unused CMake variable.
 if (CMAKE_CROSSCOMPILING_EMULATOR)
 endif()
+
+# TODO: CMake appends <sysroot>/usr/include to implicit includes; switching to use usr/include will make this redundant.
+set(_em_sysroot_include "${EMSCRIPTEN_SYSROOT}/include")
+if("${CMAKE_VERSION}" VERSION_LESS "3.20")
+  file(TO_CMAKE_PATH "${_em_sysroot_include}" _em_sysroot_include)
+else()
+  cmake_path(CONVERT "${_em_sysroot_include}" TO_CMAKE_PATH_LIST _em_sysroot_include)
+endif()
+if ("${CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES}" STREQUAL "")
+  set(CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES "${_em_sysroot_include}")
+endif()
+if ("${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES}" STREQUAL "")
+  set(CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES "${_em_sysroot_include}")
+endif()
```

Comment:
Why not unconditionally append to this list?

Reply:
I didn't look into what's causing it but for me the toolchain file is interpreted three times resulting in three same paths in the list. I wanted to avoid it.